src/script/descriptor.cpp

```diff
@@ -536,6 +543,65 @@ std::unique_ptr<Descriptor> ParseScript(Span<const char>& sp, ParseScriptContext
     return nullptr;
 }
 
+
+std::unique_ptr<PubkeyProvider> InferPubkey(const CPubKey& pubkey, ParseScriptContext, const SigningProvider& provider)
+{
+    std::unique_ptr<PubkeyProvider> key_provider = MakeUnique<ConstPubkeyProvider>(pubkey);
+    return key_provider;
+}
+
+
+std::unique_ptr<Descriptor> InferScript(const CScript& script, ParseScriptContext ctx, const SigningProvider& provider)
+{
+    std::vector<std::vector<unsigned char>> data;
+    txnouttype txntype;
+
+    if(!Solver(script, txntype, data))
+        return MakeUnique<RawDescriptor>(script);
```

Comment:
More differences. @UdjinM6 Of note, we should should at least follow style guide, so space between if and paren, and use brackets.

Reply:
Done